File: API/Entities/Invoice.cs
using System.ComponentModel.DataAnnotations;

namespace API.Entities
{
    public class Invoice:BaseEntity
    {
        [Required]
        public Guid CustomerId { get; set; }
        public virtual Customer Customer { get; set; }
        [Required]
        public string InvoiceNumber { get; set; }
        public string OrderNumber { get; set; }
        [Required]
        public DateTime InvoiceDate { get; set; }
        public Guid? TermsId { get; set; }
        public virtual Terms Terms { get; set; }
        public DateTime DueDate { get; set; }
        public Guid? SalesPersonId { get; set; }
        public virtual SalesPerson SalesPerson { get; set; }
        public string Subject { get; set; }
        public string Currency { get; set; }
        public ICollection<InvoiceItem> Items { get; set; } = new HashSet<InvoiceItem>();
        public decimal SubTotal { get; set; }
        public int Discount { get; set; }
        public decimal Adjustments { get; set; }
        public string CustomerNotes { get; set; }
        public string TermsAndConditions { get; set; }
        public string AttachFile { get; set; } 
        public Income Income { get; set; }
        public decimal Total { get; set; }
        public Boolean Paid { get; set; }

    }
}

File: Client/src/app/_model/invoice.ts
import { Guid } from 'guid-typescript';
import { Customer } from './customer';
import { Income } from './income';
import { InvoiceItem } from './invoiceItem';
import { SalesPerson } from './salesPerson';
import { Tax } from './tax';
import { Term } from './term';

export interface Invoice {
  id: Guid;
  customer: Customer;
  invoiceNumber: string;
  orderNumber: string;
  invoiceDate: Date;
  terms: Term;
  dueDate: Date;
  salesPerson: SalesPerson;
  subject: string;
  currency: string;
  items: InvoiceItem[];
  subTotal: number;
  discount: number;
  adjustments: number;
  customerNotes: string;
  termsAndConditions: string;
  attachFile: string;
  income: Income;
  paid: Boolean;
  total: number;
}
